benchmark/definition/benchmarking.py

```diff
@@ -0,0 +1,42 @@
+"""Definition of the Benchmark class."""
+
+import time
+from collections.abc import Callable
+from uuid import UUID, uuid4
+
+from attrs import define, field
+from attrs.validators import instance_of
+from pandas import DataFrame
+
+from benchmark.result.result import Result
+
+
+@define
+class Benchmark:
+    """A class to define a benchmark task."""
+
+    name: str = field(validator=instance_of(str))
+    """The name of the benchmark."""
+
+    benchmark_function: Callable[[], tuple[DataFrame, dict[str, str]]]
```

Comment:
The type hint here just specifies that the callable should not accept any arguments? Or what does this say?

Reply:
That's exactly what it means.